cmake/EthCompilerSettings.cmake

```diff
@@ -33,10 +33,10 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 	add_compile_options(-Werror)
 
 	# Configuration-specific compiler settings.
-	set(CMAKE_CXX_FLAGS_DEBUG          "-O0 -g -DETH_DEBUG")
+	set(CMAKE_CXX_FLAGS_DEBUG          "-O0 -g3 -DETH_DEBUG")
 	set(CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG")
 	set(CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG")
```

Comment:
Do we still need `ETH_DEBUG` and `NDEBUG`? We should just remove them entirely.

Reply:
agree!

Reply:
Ok. As a general rule though since both of you know more about what your talking about than me, I think it would be better for you all to just add commits here rather than doing this and possibly adding noise and getting it wrong.